java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogSnapshotMetadataStore.java

```diff
@@ -216,6 +220,49 @@ private static TransactionLogSnapshotMetadata getSnapshotFromItem(Map<String, At
         return new TransactionLogSnapshotMetadata(getStringAttribute(item, PATH), type, getLongAttribute(item, TRANSACTION_NUMBER, 0));
     }
 
+    /**
+     * Retrieves metadata of snapshots older than a certain time. Note that this excludes latest snapshots.
+     *
+     * @param  time the time used to decide which snapshots are oldest
+     * @return      a stream of snapshots that were last updated before the provided time
+     */
+    public Stream<TransactionLogSnapshotMetadata> getOldestSnapshots(Instant time) {
```

Comment:
This name could be more specific, you need to read a bit to work out what this is doing. I'd decide between getExpiredSnapshotsAtTime, or getSnapshotsBefore and pass in the time that's currently called "expiryDate". If we go with the second, we wouldn't need the expiry time in this class.